samples/net/mqtt/src/modules/network/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# LTE boards
+target_sources_ifdef(CONFIG_BOARD_NRF9160DK_NRF9160_NS
+			app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/network_lte.c)
+
+target_sources_ifdef(CONFIG_BOARD_THINGY91_NRF9160_NS
```

Comment:
You could make the source inclusion based on `CONFIG_NRF_MODEM_LIB` to have only one